ethstaker_deposit/utils/click.py

```diff
@@ -120,4 +120,15 @@ def choice_prompt_func(prompt_func: Callable[[], str], choices: Sequence[str]) -
     '''
     Formats the prompt and choices in a printable manner.
     '''
-    return lambda: '%s %s: ' % (prompt_func(), choices)
+    # A join with unconditional embedded LTR can add non-printing characters on some Terminals
+    # Iterate over choices instead and use LTR embedding if the string has RTL embedding
+    output = '['
+    for i in range(len(choices)):
+        output = output + choices[i]
+        if i < len(choices) - 1:
+            if '\u202b' in choices[i]:
+                output = output + '\u202a, \u202c'
```

Comment:
To fix the comma issue, which I see as well, would the fix be
```suggestion
                output = output + '\u202a \u202c, '
```

`Please choose your language [1. ‫العربية‬‪, ‬2. ελληνικά, 3. Eng`

vs

`Please choose your language [1. ‫العربية‬‪ ‬, 2. ελληνικά, 3. Eng`

Reply:
Yes, which is utterly non-intuitive to me. RTL breaks my brain :sweat_smile: . On my terminal I need another space so there's one between the command and "2."